servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyHttpServer.java

```diff
@@ -278,38 +269,7 @@ static final class NettyHttpServerConnection extends HttpServiceContext implemen
                     connection.executionContext().ioExecutor(), connection.executionContext().executor(),
                     HttpExecutionStrategies.offloadNone());
             this.service = service;
-            flushStrategy = new SplittingFlushStrategy(connection.defaultFlushStrategy(),
-                    // h2 may return a single HttpResponseMetaData for an empty response in some scenarios,
-                    // otherwise a trailers object will be included to indicate the end because content-length isn't
-                    // mutually exclusive from trailers in h2.
-                    protocol().major() > 1 ?
-                            itemWritten -> (itemWritten instanceof HttpResponseMetaData &&
-                                        emptyMessageBody((HttpResponseMetaData) itemWritten)) ||
-                                    itemWritten instanceof HttpHeaders ? End : InProgress :
-                    new FlushBoundaryProvider() {
-                        private long contentLength;
-                        @Override
-                        public FlushBoundary detectBoundary(@Nullable final Object itemWritten) {
-                            if (itemWritten instanceof HttpResponseMetaData) {
-                                final HttpResponseMetaData metadata = (HttpResponseMetaData) itemWritten;
-                                contentLength = getContentLength(metadata);
-                                // The content length maybe unknown at this point (e.g. 204 response) but then later
-                                // determined to be 0. In that case we should conservatively use End and rely upon
-                                // adjustForMissingBoundaries to accommodate if more data comes. Otherwise the
-                                // FlushStrategy may not trigger a flush (e.g. flushOnEnd) and if so the response
-                                // won't actually be written.
-                                return contentLength > 0 || (contentLength < 0 &&
-                                        (!emptyMessageBody(metadata) &&
-                                                isTransferEncodingChunked(metadata.headers()))) ? Start : End;
-                            }
-                            if (itemWritten instanceof Buffer) {
-                                return contentLength > 0 &&
-                                        (contentLength -= ((Buffer) itemWritten).readableBytes()) <= 0 ?
-                                        End : InProgress;
-                            }
-                            return itemWritten instanceof HttpHeaders ? End : InProgress;
-                        }
-                    });
+            this.flushStrategy = new ChangingFlushStrategy(new FlushStrategyHolder(connection.defaultFlushStrategy()));
             connection.updateFlushStrategy((current, isCurrentOriginal) -> flushStrategy);
```

Comment:
>Temporarily introduce ChangingFlushStrategy to still support changes of the FlushStrategy per request with existing FlushStrategy API;

can you clarify what your plan is for removal of flush strategy on the connection level? is this planned for followup PR or to be included in this PR?

very happy to see `SplittingFlushStrategy` and related hacks go 🏅